src/devinterp/slt/norms.py

```diff
@@ -0,0 +1,29 @@
+import torch
+import torch.nn as nn
+
+from devinterp.slt.callback import ChainCallback
+
+class WeightNorm(ChainCallback):
+    def __init__(self, num_chains, num_draws, device='cpu'):
+        self.weight_norms = torch.tensor((num_chains, num_draws), dtype=torch.float32).to(device)
+        self.device = device
+    
+    def __call__(self, chain: int, draw: int, model: nn.Module):
+        self.update(chain, draw, model)
+    
+    def update(self, chain: int, draw: int, model: nn.Module):
+        weights = self._flatten_weights(model)
+        self.weight_norms[chain, draw] = torch.linalg.vector_norm(weights, ord=2)
+
+    def sample(self):
+        return {
+            "weight_norm/trace": self.weight_norms.cpu().numpy()
+        }
+    
+    def _flatten_weights(self, model: nn.Module):
```

Comment:
Same feedback again as on #49 and #48. Probably better to avoid allocating this tensor